symbol not found in flat namespace '__end'

#161 is back again, this time on the execution side (and only on mac). From https://github.com/spacemeshos/go-spacemesh/actions/runs/11674170614/job/32506374469?pr=6380:

```
=== FAIL: node TestSpacemeshApp_TransactionService (0.49s)
panic: loading Athena VM: loading library: dlopen(/Users/m1/actions-runner/_work/go-spacemesh/go-spacemesh/build/libathenavmwrapper.dylib, 0x000A): symbol not found in flat namespace '__end' [recovered]
	panic: loading Athena VM: loading library: dlopen(/Users/m1/actions-runner/_work/go-spacemesh/go-spacemesh/build/libathenavmwrapper.dylib, 0x000A): symbol not found in flat namespace '__end'

goroutine 1817 [running]:
testing.tRunner.func1.2({0x106fe5ce0, 0xc0010a4280})
	/Users/m1/actions-runner/_work/_tool/go/1.23.2/arm64/src/testing/testing.go:1632 +0x2c4
testing.tRunner.func1()
	/Users/m1/actions-runner/_work/_tool/go/1.23.2/arm64/src/testing/testing.go:1635 +0x47c
panic({0x106fe5ce0?, 0xc0010a4280?})
	/Users/m1/actions-runner/_work/_tool/go/1.23.2/arm64/src/runtime/panic.go:785 +0x124
github.com/spacemeshos/go-spacemesh/vm/sdk/wallet.Spawn({0xc00059ecc0, 0x40, 0x40}, 0x0, {0xc0006cfe70, 0x1, 0x0?})
	/Users/m1/actions-runner/_work/go-spacemesh/go-spacemesh/vm/sdk/wallet/tx.go:45 +0x5d0
github.com/spacemeshos/go-spacemesh/node.TestSpacemeshApp_TransactionService(0xc00029b[380](https://github.com/spacemeshos/go-spacemesh/actions/runs/11674170614/job/32506374469?pr=6380#step:14:381))
	/Users/m1/actions-runner/_work/go-spacemesh/go-spacemesh/node/node_test.go:554 +0xc98
testing.tRunner(0xc00029b380, 0x1072b89f8)
	/Users/m1/actions-runner/_work/_tool/go/1.23.2/arm64/src/testing/testing.go:1690 +0x188
created by testing.(*T).Run in goroutine 1
	/Users/m1/actions-runner/_work/_tool/go/1.23.2/arm64/src/testing/testing.go:1743 +0x5e4
```

--- a/vm/entrypoint/src/syscalls/memory.rs
+++ b/vm/entrypoint/src/syscalls/memory.rs
@@ -16,9 +16,10 @@
 // Note: We inherit this constraint from SP1, and I don't see a reason to remove it.
 const MAX_MEMORY: usize = 0x78000000;
 
-#[allow(clippy::missing_safety_doc)]
-#[no_mangle]
-pub unsafe extern "C" fn sys_alloc_aligned(bytes: usize, align: usize) -> *mut u8 {
+/// # Safety
+/// - `align` must not be zero and must be a power of 2
+/// - `bytes` must not be zero
+pub unsafe fn sys_alloc_aligned(bytes: usize, align: usize) -> *mut u8 {
   extern "C" {
     // https://lld.llvm.org/ELF/linker_script.html#sections-command
     static _end: u8;

--- a/vm/lib/src/lib.rs
+++ b/vm/lib/src/lib.rs
@@ -9,5 +9,4 @@ extern "C" {
   pub fn syscall_read(fd: u32, read_buf: *mut u8, nbytes: usize);
   pub fn syscall_hint_len() -> usize;
   pub fn syscall_hint_read(ptr: *mut u8, len: usize);
-  pub fn sys_alloc_aligned(bytes: usize, align: usize) -> *mut u8;
 }

